diff --git a/pthread.c b/pthread.c
index 13f15ba..6423416 100644
--- a/pthread.c
+++ b/pthread.c
@@ -199,6 +199,26 @@ int pthread_attr_getstackaddr(const pthread_attr_t *attr, void **stackaddr)
     return OK;
 }
 
+int pthread_attr_setstack(pthread_attr_t *attr, void *stackaddr, size_t stacksize)
+{
+    // FIXME these operations should probably be transactional
+    if (!pthread_attr_setstackaddr(attr, stackaddr))
+        return errno;
+    if (!pthread_attr_setstacksize(attr, stacksize))
+        return errno;
+    return OK;
+}
+
+int pthread_attr_getstack(const pthread_attr_t *attr, void **stackaddr, size_t *stacksize)
+{
+    // FIXME these operations should probably be transactional
+    if (!pthread_attr_getstackaddr(attr, stackaddr))
+        return errno;
+    if (!pthread_attr_getstacksize(attr, stacksize))
+        return errno;
+    return OK;
+}
+
 int pthread_attr_setdetachstate(pthread_attr_t *attr, int detachstate)
 {
     int s;
@@ -283,6 +303,17 @@ int pthread_attr_getprio_np(const pthread_attr_t *attr, int *prio)
     return OK;
 }
 
+int pthread_getattr_np(pthread_t thread, pthread_attr_t *attr)
+{
+    pth_attr_t pthattr = pth_attr_of((pth_t)thread);
+    if (!pthattr)
+        return -1;
+    memcpy(attr, pthattr, sizeof(*attr));
+    free(pthattr);
+    return 0;
+}
+
+
 /*
 **  THREAD ROUTINES
 */
diff --git a/pthread.h.in b/pthread.h.in
index 54c921c..88f8b7e 100644
--- a/pthread.h.in
+++ b/pthread.h.in
@@ -334,6 +334,8 @@ extern int       pthread_attr_setstacksize(pthread_attr_t *, size_t);
 extern int       pthread_attr_getstacksize(const pthread_attr_t *, size_t *);
 extern int       pthread_attr_setstackaddr(pthread_attr_t *, void *);
 extern int       pthread_attr_getstackaddr(const pthread_attr_t *, void **);
+extern int       pthread_attr_setstack(pthread_attr_t *, void *, size_t);
+extern int       pthread_attr_getstack(const pthread_attr_t *, void **, size_t *);
 extern int       pthread_attr_setdetachstate(pthread_attr_t *, int);
 extern int       pthread_attr_getdetachstate(const pthread_attr_t *, int *);
 extern int       pthread_attr_setguardsize(pthread_attr_t *, int);
@@ -342,6 +344,7 @@ extern int       pthread_attr_setname_np(pthread_attr_t *, char *);
 extern int       pthread_attr_getname_np(const pthread_attr_t *, char **);
 extern int       pthread_attr_setprio_np(pthread_attr_t *, int);
 extern int       pthread_attr_getprio_np(const pthread_attr_t *, int *);
+extern int       pthread_getattr_np(pthread_t, pthread_attr_t *);
 
 /* thread routines */
 extern int       pthread_create(pthread_t *, const pthread_attr_t *, void *(*)(void *), void *);
